src/watcher.js

```diff
@@ -266,6 +267,40 @@ class Watcher extends SafeEventEmitter {
         this.on('load.chat', updateChannel);
         this.on('load.vod', updateChannel);
     }
+
+    tmiEventObserver() {
+        const PATCHED_SENTINEL = () => {};
+        let twitchConsumeEvent;
+        const watcher = this;
+        function bttvConsumeEvent(event) {
+            let prevent = false;
+            event.preventDefault = () => {
```

Comment:
you shouldn't mutate Twitch's record here, you should instead copy it for our internal use.

also more performant to not keep creating a new function on each event..

since the logic of this observer is more complicated than the rest and this file is getting large, perhaps it's time to break out into a new `watchers` folder.